src/Sensor/SensorOptical.cpp

```diff
@@ -11,6 +11,8 @@
 #define ROLLER_RADIUS 0.08276057 // metres
 #define VELOCITY_FACTOR (GEAR_RATIO * ROLLER_RADIUS) // speed of vehicle [m/s] = velocity factor * angular velocity [rad/s]
 
+#define UINT32_T_MAX 0xffffffffU
```

Comment:
It seems like there is already a definition for ```UINT32_MAX```, but it expands to ```0xffffffffUL``` which I believe would be a 64 bit variable. As mentioned in issue #62 , this is not desirable

Reply:
```unsigned long``` is actually a 32-bit type.  Like I said before, the size of C types vary depending on the architecture.  If you were compiling for a 64-bit CPU, ```unsigned long``` would be a 64 bits, but for the pico it is still 32 bits.  I think the C type for a 64-bit unsigned int on the pico would be ```unsigned long long```.  Just another example of why in our own code we use the explicitly-defined types in <stdint.h>.

In general, I would trust the standard library implementations of all these things, as a lot of work goes into making sure that they are correct for a given architecture.  We should trust that all the min and max values defined in <stdint.h> are the correct values/types and all that.

When in doubt, you can print out the size of a type:

```
Serial.println("sizeof(unsigned long): " + String(sizeof(unsigned long));

// prints
sizeof(unsigned long): 4
```

Reply:
If you want to go deeper, you can navigate to the file where all these are defined:

```C:/<user>/.platformio/packages/toolchain-rp2040-earlephilhower/arm-none-eabi/include/stdint.h```

These numeric type definitions and additional macros are part of the arm implementation of the C standard library, designed for the architecture that we're using and included in the platformIO arduino-pico dev kit.

Reply:
Sounds good, I have changed it so we are using the system defined ```UINT32_MAX``` macro.

Reply:
I have added some unit tests, and my tests did not expose any issues with the logic.

I tested the two cases where micros overflowed and the readInterval time had elapsed since the last update time (in which case angular velocity should be calculated), and two cases where the readInterval time had not elapsed since the last update time (in which case angular velocity should not have been calculated).